### PR TITLE
added tunnel plugin to krew

### PR DESCRIPTION
**Pull request by omrikiei**
Added a plugin manifest for a tunnel([ktunnel](https://github.com/omrikiei/ktunnel))

### CODE REVIEW

File: plugins/tunnel.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: tunnel
+spec:
+  version: v1.3.8
+  homepage: https://github.com/omrikiei/ktunnel
+  shortDescription: Establishes a reverse tunnel between a kubernetes cluster and your local machine
```
Comment (ahmetb) [inline plugins/tunnel.yaml:8]: Please shorten to fit ~65 chars.

File: plugins/tunnel.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  description: |
+    This plugin establishes a reverse tunnel between a kubernetes cluster and your local machine. 
+    It lets you expose your machine as a service in the cluster or expose it to a specific deployment as a local host.
```
Comment (ahmetb) [inline plugins/tunnel.yaml:11]: please line wrap at 80 chars.
Kubernetes capitalized.

**Comment by ahmetb** [comment]
The name `tunnel` is a common word, but not so common in Kubernetes context. I think it is fine to assign to first-comer in this case. @corneliusweig @chriskim06 thoughts?

**Comment by chriskim06** [comment]
I second that, the name seems fine to me. Cool plugin btw 😄

File: plugins/tunnel.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+metadata:
+  name: tunnel
+spec:
+  version: v1.3.8
+  homepage: https://github.com/omrikiei/ktunnel
+  shortDescription: Establish a reverse tunnel between Kubernetes your local machine
+  description: |
+      This plugin establishes a reverse tunnel between Kubernetes and your local
+machine. It lets you expose your machine as a service in the cluster or expose it to a
+specific deployment as a local host.
```
Comment (ahmetb) [inline plugins/tunnel.yaml:12]: I think you need to indent this in. see other manifests as an example.

File: plugins/tunnel.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: tunnel
+spec:
+  version: v1.3.8
+  homepage: https://github.com/omrikiei/ktunnel
+  shortDescription: Establish a reverse tunnel between Kubernetes your local machine
```
Comment (ahmetb) [inline plugins/tunnel.yaml:8]: I will make this shorter for you:

> Reverse tunneling between cluster and your machine

**Review by ahmetb — CHANGES_REQUESTED** [review]

**Comment by corneliusweig** [comment]
I haven't looked carefully yet, so just my 2cent for now:

- We already have a few plugins for tunneling/forwarding/proxying requests. I think it would make sense to pick a more descriptive name, such as `reverse-tunnel` (so something with reverse, because this seems to be the main use case)
- telemetry.io is a pretty popular project for the same purpose. How does your plugin compare to that solution?

**Comment by omrikiei** [comment]
> I haven't looked carefully yet, so just my 2cent for now:
> 
> * We already have a few plugins for tunneling/forwarding/proxying requests. I think it would make sense to pick a more descriptive name, such as `reverse-tunnel` (so something with reverse, because this seems to be the main use case)
> * telemetry.io is a pretty popular project for the same purpose. How does your plugin compare to that solution?

Hey @corneliusweig, which other tunneling plugins are listed? I'd like to take a look. could you also share some reference to telemetry.io? I wasn't able to find anything in that context.. did you mean telepresence.io?

**Comment by corneliusweig** [comment]
Meh.. I should not type on phone  :). Sorry for the confusion.

You're right, I meant telepresence.io. The other tunnel plugins are
- net-forward
- socks5-proxy
- tap

**Comment by omrikiei** [comment]
telepresense substitutes a two-way network proxy for your normal pod running in the Kubernetes cluster.. it would start a dedicated shell and proxy all the resource coupled with the deployment to that shell.
ktunnel is a reverse network proxy and has only this functionality - it can be used for various use cases like [remote debugging](https://github.com/omrikiei/ktunnel/tree/master/examples/pycharm-remote-debugging-on-k8s#readme) (which is why I originally built it), metrics population, file transfer and I've seen [other cool usages as well](http://mbuffett.com/posts/ktunnel-ngrok-replace/).
ktunnel is not only limited for use in kubernetes, and it doesn't replace any remote resources with local ones

File: plugins/tunnel.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+metadata:
+  name: tunnel
+spec:
+  version: v1.3.8
+  homepage: https://github.com/omrikiei/ktunnel
+  shortDescription: Reverse tunneling between cluster and your machine
+  description: |
+      This plugin establishes a reverse tunnel between Kubernetes and your local machine. 
+      It lets you expose your machine as a service in the cluster or expose it to a
+specific deployment as a local host.
```
Comment (corneliusweig) [inline plugins/tunnel.yaml:12]:
```suggestion
      This plugin establishes a reverse tunnel between the Kubernetes cluster
      and your local machine. It lets you expose your machine as a service in
      the cluster or expose it to a specific deployment as a local host.
```

**Review by corneliusweig — COMMENTED** [review]
Ok, thanks for the explanation. Let's add this then. I have a minor wording (and text wrapping) suggestion, otherwise this is good to go!

Thanks for contributing!
/approve

---

Btw, you can shorten https://github.com/omrikiei/ktunnel/blob/6a5d8af7c979f877d63153932c0e032c410bea59/pkg/k8s/common.go#L25-L28 to
```go
_ "k8s.io/client-go/plugin/pkg/client/auth"
```
Then you will also get additional auth provider when they are added.

**Comment by omrikiei** [comment]
> https://github.com/omrikiei/ktunnel/blob/6a5d8af7c979f877d63153932c0e032c410bea59/pkg/k8s/common.go#L25-L28

I like the way you think

**Comment by corneliusweig** [comment]
/lgtm
/approve
Thanks for contributing!

**Comment by ahmetb** [comment]
Also, please consider setting up [Krew release automation](https://krew.sigs.k8s.io/docs/developer-guide/release/automating-updates/) which helps you skip manually making updates to your Krew manifests on each new version and send a pull request. It’s a GitHub workflow bot that can run every time you push a new tag. Those PRs are automatically approved.

**Pull request merged**